compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -176,25 +176,50 @@ uint8_t getMemoryBarrierBinaryLengthLowerBound(int32_t barrier, TR::CodeGenerato
 
 // -----------------------------------------------------------------------------
 // OMR::X86::Instruction:: member functions
+bool OMR::X86::Instruction::needsRepPrefix()
+   {
+   return self()->getOpCode().needsRepPrefix();
+   }
 
-uint8_t *OMR::X86::Instruction::generateBinaryEncoding()
+bool OMR::X86::Instruction::needsLockPrefix()
    {
-   uint8_t *instructionStart = self()->cg()->getBinaryBufferCursor();
-   uint8_t *cursor           = instructionStart;
-   if (self()->getOpCode().needsRepPrefix())
+   return self()->getOpCode().needsLockPrefix();
+   }
+
+uint8_t* OMR::X86::Instruction::generateBinaryEncoding()
+   {
+   // Normally the loop only executes once,
+   // it is needed to avoid recursive calls when operand() requests to regenerate the binary code by returning NULL;
+   for(;;)
       {
-      *cursor++ = 0xf3;
+      uint8_t *instructionStart = self()->cg()->getBinaryBufferCursor();
+      uint8_t *cursor           = instructionStart;
+      self()->setBinaryEncoding(instructionStart);
+      if (self()->needsRepPrefix())
+         {
+         *cursor++ = 0xf3;
+         }
+      if (self()->needsLockPrefix())
+         {
+         *cursor++ = 0xf0;
+         }
+      cursor = self()->generateRepeatedRexPrefix(cursor);
+      cursor = self()->getOpCode().binary(cursor, self()->rexBits());
+      cursor = self()->generateOperand(cursor);
+      // cursor is normal not NULL, and hence we can finish binary encoding and exit the loop
+      // cursor is NULL when operand() requests to regenerate the binary code, which may happen during encoding of memref with unresolved symbols on 64-bit
```

Comment:
This is an outdated diff. The comment containing operand() has been updated. Failing to renaming will trigger a compile-time error. This was the only function that refers operand() in comments.